health/notifications/alarm-notify.sh.in

```diff
@@ -1664,6 +1675,55 @@ send_awssns() {
 	return 1
 }
 
+# -----------------------------------------------------------------------------
+# matrix sender
+
+send_matrix() {
+    local homeserver="${1}" webhook accesstoken rooms="${2}" httpcode sent=0 payload color
+
+	[ "${SEND_MATRIX}" != "YES" ] && return 1
+
+    # check if access token was provided
+    if [ "${SEND_MATRIX}" == "YES"  ] && [ -n "${MATRIX_ACCESSTOKEN}" ]; then
+        accesstoken="${MATRIX_ACCESSTOKEN}"
+    else
+        return 1
+    fi
+
+	case "${status}" in
+	WARNING) color="warning" ;;
+	CRITICAL) color="danger" ;;
+	CLEAR) color="good" ;;
+	*) color="#777777" ;;
+	esac
+
+	for room in ${rooms}; do
+        webhook="$homeserver/_matrix/client/r0/rooms/$(urlencode $room)/send/m.room.message?access_token=$accesstoken"
+		payload="$(
+        cat <<EOF
+        {
+            "msgtype": "m.notice",
+            "format": "org.matrix.custom.html",
+            "formatted_body": "<font color=\"$color\">${host} ${status_message}, \`${chart}\` (${family}), <b>${alarm}</b></font>",
+            "body": "${host} ${status_message}, \`${chart}\` (${family}), *${alarm}*"
```

Comment:
Can you add a link to ${goto_url} somewhere here? In a notification I received I saw (surprisingly) that family was converted to a link that leads nowhere. I didn't go any deeper into the documentation of the API, but it shouldn't be like that. Perhaps make the host or the alarm clickable and pass goto_url. You don't need to urlencode it, it's already ready for use. You can check other examples.